cmd/crossplane/core/core.go

```diff
@@ -226,19 +227,25 @@ func (c *startCommand) Run(s *runtime.Scheme, log logging.Logger) error { //noli
 	if c.EnableCompositionFunctions {
 		o.Features.Enable(features.EnableBetaCompositionFunctions)
 		log.Info("Beta feature enabled", "flag", features.EnableBetaCompositionFunctions)
+
 		clienttls, err := certificates.LoadMTLSConfig(
 			filepath.Join(c.TLSClientCertsDir, initializer.SecretKeyCACert),
 			filepath.Join(c.TLSClientCertsDir, corev1.TLSCertKey),
 			filepath.Join(c.TLSClientCertsDir, corev1.TLSPrivateKeyKey),
 			false)
-
 		if err != nil {
 			return errors.Wrap(err, "cannot load client TLS certificates")
 		}
+
+		m := xfn.NewMetrics()
+		metrics.Registry.MustRegister(m)
```

Comment:
metrics registration should go to an init func. Otherwise, you will never be able to create multiple instances of this without a panic.

Reply:
I don't think it's that cut-and-dry, and would generally prefer to setup things like metrics, loggers etc here in "main" [^1] and plumb them down rather than relying on init functions hidden away in individual packages.

[^1]: This isn't literally `func main` but it's close enough in practice. I can't see us unit testing this plumbing.